File: 프로그래머스/1/133502. 햄버거 만들기/햄버거 만들기.cpp
#include <string>
#include <vector>
using namespace std;

vector<int> stack;

int solution(vector<int> ingredient) {
    // 빵 - 야채 - 고기 - 빵
    // 1 - 빵
    // 2 - 야채
    // 3 - 고기
    int answer = 0;
    
    for(int i=0;i<ingredient.size();i++){
        if(stack.size()<2){
            stack.push_back(ingredient[i]);
        }
        else{
            stack.push_back(ingredient[i]);
            if(stack[stack.size()-4]==1 && stack[stack.size()-3]==2 && stack[stack.size()-2]==3 && stack[stack.size()-1]==1){
                stack.pop_back();
                stack.pop_back();
                stack.pop_back();
                stack.pop_back();
                answer++;
            }
        }
    }
    
    return answer;
}

File: 프로그래머스/1/133502. 햄버거 만들기/햄버거 만들기.java
import java.util.*;
class Solution {
    public int solution(int[] ingredient) {
        int answer = 0;
        Stack<Integer> stack = new Stack<>();
        for(int item: ingredient){
            if(stack.empty()){
                stack.push(item);
            }
            else{
                stack.push(item);
                if(stack.size()>3 && stack.elementAt(stack.size()-1)==1 && stack.elementAt(stack.size()-2)==3 && stack.elementAt(stack.size()-3)==2 && stack.elementAt(stack.size()-4)==1){
                    stack.pop();
                    stack.pop();
                    stack.pop();
                    stack.pop();
                    answer++;
                }
            }
        }
        return answer;
    }
}